src/idl_gen_cpp.cpp

```diff
@@ -3068,7 +3071,8 @@ class CppGenerator : public BaseGenerator {
             #if FLATBUFFERS_CPP_OBJECT_UNPACKTO
             const bool is_pointer =
                 field.value.type.VectorType().base_type == BASE_TYPE_STRUCT &&
-                !IsStruct(field.value.type.VectorType());
+                !IsStruct(field.value.type.VectorType()) &&
```

Comment:
Same with this, it would be easier to understand if this was a static method with a IsVectorOfTables() or something.